modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/GlobalSnapshotAcceptanceManager.scala

```diff
@@ -73,14 +74,16 @@ object GlobalSnapshotAcceptanceManager {
       for {
         acceptanceResult <- acceptBlocks(blocksForAcceptance, lastSnapshotContext, lastActiveTips, lastDeprecatedTips, ordinal)
 
-        (scSnapshots, currencySnapshots, returnedSCEvents) <- stateChannelEventsProcessor.process(
-          ordinal,
-          lastSnapshotContext,
-          scEvents,
-          validationType
-        )
+        StateChannelAcceptanceResult(scSnapshots, currencySnapshots, returnedSCEvents, currencyAcceptanceBalanceUpdate) <-
+          stateChannelEventsProcessor
+            .process(
+              ordinal,
+              lastSnapshotContext.copy(balances = lastSnapshotContext.balances ++ acceptanceResult.contextUpdate.balances),
+              scEvents,
+              validationType
+            )
         sCSnapshotHashes <- scSnapshots.toList.traverse {
-          case (address, nel) => nel.head.toHashed.map(address -> _.hash)
+          case (address, nel) => nel.last.toHashed.map(address -> _.hash)
```

Comment:
I know it's generally non-performant to get the last element of a list like that. I tried changing the collection to be NonEmptyChain or Chain, but in the end decided that it makes no sense at this moment. We start the processing with a list and we need to end with a list (because of schema), if we transform to NEC and back from NEC to have the correct type we may add more boilerplate and end up with the same complexity - we can always change it in the future if decide it makes sense. For the most cases it won't matter much as we won't work with long lists here - 1, 2 elements is probably what we have in those.

Reply:
@TheMMaciek I think this is a good case for a comment so that future readers won't question it.

Reply:
It is used in multiple places, not sure if explaining it in one makes sense. Wouldn't like to pollute code with comments as well if they are not strictly neccessary.